keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,951 @@
+  - Handling updates to the OCI image or artifact, similar to how ConfigMaps update mounted files.
+
+The following code snippet illustrates the proposed API change:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example-pod
+spec:
+  volumes:
+  - name: oci-volume
+    oci:
+      image: "example.com/my-image:latest"
+  containers:
+  - name: my-container
+    image: busybox
+    volumeMounts:
+    - mountPath: /data
+      name: oci-volume
+```
+
+### Kubelet and Container Runtime Interface (CRI) support for OCI artifacts
+
+Kubelet and the Container Runtime Interface (CRI) currently handle OCI images. To support OCI artifacts,
+potential enhancements may be required:
+
+**Extended Media Type Handling:**
+   - Update Kubelet and CRI to recognize and handle new media types associated with OCI artifacts.
+   - Ensure that pulling and storing these artifacts is as efficient and secure as with OCI images.
+
+**Lifecycling and Garbage Collection:**
+   - Implement custom logic for managing the lifecycle of various OCI artifacts.
+
+**Artifact-Specific Configuration:**
+   - Introduce new configuration options or CRD to handle the unique requirements of different types of OCI artifacts.
+
+**Validation:**
+   - Extend validation and security checks to cover new artifact types.
+
+**Storage Optimization:**
+   - Develop optimized storage solutions tailored for different artifact types,
+     potentially integrating with existing storage solutions or introducing new mechanisms.
+
+#### kubelet
+
+Some parts of the existing kubelet code can be reused, for example:
+
+- The logic how to ensure that an image exists on the node:
+  https://github.com/kubernetes/kubernetes/blob/39c6bc3/pkg/kubelet/images/image_manager.go#L102
+- The retrieval of available secrets for a pod:
+  https://github.com/kubernetes/kubernetes/blob/39c6bc3/pkg/kubelet/kubelet_pods.go#L988
+
+We consider to refactor the logic into re-usable bits for both, the kubelet as
+well as the volume plugin, because volume plugins run in a dedicated go routine
+at the beginning of the kubelet initialization:
+
+https://github.com/kubernetes/kubernetes/blob/39c6bc3/pkg/kubelet/kubelet.go#L1633-L1634
+
+Specifying a pull policy will not be supported in the alpha implementation of the
+feature and will align to `PullAlways`.
+
+For registry authentication purposes the same credentials will be used as for
+the running pod, if available.the
+
+#### CRI
+
+The CRI is already capable of pulling images [via the `PullImage` RPC](https://github.com/kubernetes/cri-api/blob/3a66d9d/pkg/apis/runtime/v1/api.proto#L154).
+This does not necessarily include OCI artifacts and is dependent on the used
+container runtime (CRI-O, containerd). Beside that, volume plugins usually
+require a mount from a local directory. This means we extend the CRI protocol by
+adding a two new methods as well as their request and response types:
+
+```protobuf
+// ImageService defines the public APIs for managing images.
```

Comment:
should we pass the runtime handler and pod annotations to align with what we do for regular images